### PR TITLE
Break down each deployment to it's own config file

### PR DESCRIPTION
**Pull request by chaodaiG**
Fix: #1164 
/cc @adrcunha 
/cc @coryrc

### CODE REVIEW

**Comment by srinivashegde86** [comment]
[nit] Should the files be called <type>_deployment.yaml? or they can be just called <type>.yaml considering they are under deployments dir

**Comment by chaodaiG** [comment]
> [nit] Should the files be called `<type>_deployment.yaml`? or they can be just called `<type>.yaml` considering they are under deployments dir

I think it's a good idea, done

**Comment by coryrc** [comment]
I don't think we should change the Makefile names, because then k8s.io docs will no longer be that useful. Or, if still doing so, the docs in the repo need to be updated.

**Comment by coryrc** [comment]
Oh, looks like we don't have our own copy of docs except in playbook. I don't think the Makefile names should change then.

File: ci/prow/deployments/horologuim.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 The Knative Authors
```
Comment (adrcunha) [inline ci/prow/deployments/horologuim.yaml]: typo: this file should be named horologium.yaml

File: ci/prow/deployments/prowjob-crd.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Copyright 2019 The Knative Authors
```
Comment (adrcunha) [inline ci/prow/deployments/prowjob-crd.yaml]: Maybe just crds.yaml?

File: ci/prow/Makefile [hunk 3]
```diff
@@ -45,27 +46,42 @@ update-config:
...
 	$(SET_CONTEXT)
 	kubectl create configmap plugins --from-file=plugins.yaml=plugins.yaml --dry-run -o yaml | kubectl replace configmap plugins -f -
 	$(UNSET_CONTEXT)
 
-update-boskos:
+# Update all deployments of boskos
+update-boskos-deployments:
 	$(SET_CONTEXT)
-	kubectl apply -f boskos/config.yaml
+	@for f in $(shell ls boskos/deployments/*.yaml); do kubectl apply -f $${f}; done
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply ci/prow/Makefile]: that's what I initially tried, it didn't work

File: ci/prow/Makefile [hunk 4]
```diff
@@ -45,27 +46,42 @@ update-config:
...
 	$(UNSET_CONTEXT)
 
-update-boskos-config:
+# Update single deployment of boskos, expect passing in ${NAME} like `make update-boskos-deployment NAME=janitor_deployment`
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply ci/prow/Makefile]: done, add `single` and `all` in target names for single and multiple deployments

File: ci/prow/deployments/horologuim.yaml (hunk 1 above)
Comment (chaodaiG) [reply ci/prow/deployments/horologuim.yaml]: good catch, done

File: ci/prow/deployments/prowjob-crd.yaml (hunk 2 above)
Comment (chaodaiG) [reply ci/prow/deployments/prowjob-crd.yaml]: makes sense, done

File: ci/prow/Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (chaodaiG) [reply ci/prow/Makefile]: we don't seem to have a section in any README for this Makefile, but I didn't notice some other places in `*.md` files that needs to be updated

**Pull request merged**